### ISSUE TITLE
eglot + orderless issue: choices not narrowed

### ISSUE THREAD

**Issue by jdtsmith** [oantolin/orderless#78]
Using orderless + corfu with eglot, after corfu pops up, either automatically or by invoking `M-Tab`, further typing highlights matching text, but does not narrow the list down further to matching sources.  This occurs with the orderless completion style, but not other styles such as `emacs22`.  

![image](https://user-images.githubusercontent.com/93749/131899736-e0dcbb5b-2f7d-4af9-9736-4b97a187a867.png)

This does _not_ occur in emacs-lisp modes, for example.  With a brief look, it seems as though `orderless-all-completions` always returns the same results with eglot, even as more filter terms are added.  

Minimal reproduction with `emacs -Q`:

```elisp
(require 'package)
(add-to-list 'package-archives
  	     '("melpa" . "https://melpa.org/packages/") t)
(package-initialize)
(unless package-archive-contents (package-refresh-contents))
(package-install 'corfu)
(require 'corfu)
(corfu-global-mode)
(package-install 'eglot)
(require 'eglot)
(add-hook 'python-mode-hook 'eglot-ensure)
(package-install 'orderless)
(require 'orderless)
(setq completion-styles '(orderless))
```

Then open a `.py` file and `M-Tab` to complete something, then refine the completion and notice nothing disappears, as in the shot above.  I note that eglot does "decorate" the search candidates with lots of text properties, like `"base_events" 0 1 (eglot--lsp-item (:label "base_events" :kind 9 :sortText "abase_events" :insertText #1# :data (:doc_uri "file:///Users/jdsmith/code/python/scraps/test_corfu_eldoc.py"))))`

**Comment by jdtsmith** [comment]
I wonder if this relates to the orderless workhorse `completion-regexp-list`?  That does not seem to be implemented in eglot's collection/table function, and basic completion info says (emphasis mine):

> In addition, to be acceptable, a completion must also match all the
     regular expressions in ‘completion-regexp-list’.  (Unless
     COLLECTION is a function, **in which case that function has to handle
     ‘completion-regexp-list’ itself**.)

Do you know of any collection-function based completion backends which handle `completion-regexp-list`?

**Comment by jdtsmith** [comment]
Resolved in joaotavora/eglot#733
